src/IW/Core/Url.hs

```diff
@@ -0,0 +1,8 @@
+module IW.Core.Url
+       ( Url (..)
+       ) where
+
+
+newtype Url = Url { unUrl :: Text }
+    deriving stock   (Generic, Show)
```

Comment:
Oh, also, I've just thought, maybe it's better to derive `Show` with `newtype` strategy? It would look more straightforward :thinking:

Reply:
@vrom911 I'm not really familiar with deriving strategies, but that sounds good to me. `Generic` as well?

Reply:
Deriving in `Show` case is quite visible and it can be used as a good example to display the difference between strategies. When you derive with the `stock` strategy (this is the default one and equvivalent to just `deriving Show`) then the standard instance will be generated for the data type. In case of newtypes it would look like this:

```haskell
ghci> show $ Url "abc"
Url { unUrl = "abc" }
```

`newtype` deriving strategy would produce the same instances like the ones on the underlying type (in our case as `Text` type instances). And in this case it would be:

```haskell
ghci> show $ Url "abc"
"abc"
```

Unfortunately, I don't know any good links to provide to read about this better :( Probably @chshersh could help me here.

`Generic` should be derived with `stock` strategy.

Reply:
@vrom911 You explained this nicely. Thank you.